spec.html

```diff
@@ -32938,11 +32922,11 @@ <h1>
         </dl>
         <emu-alg>
           1. If _hour_ is not finite, _min_ is not finite, _sec_ is not finite, or _ms_ is not finite, return *NaN*.
-          1. Let _h_ be 𝔽(! ToIntegerOrInfinity(_hour_)).
-          1. Let _m_ be 𝔽(! ToIntegerOrInfinity(_min_)).
-          1. Let _s_ be 𝔽(! ToIntegerOrInfinity(_sec_)).
-          1. Let _milli_ be 𝔽(! ToIntegerOrInfinity(_ms_)).
-          1. Return ((_h_ × msPerHour + _m_ × msPerMinute) + _s_ × msPerSecond) + _milli_.
+          1. Let _h_ be ! ToIntegerOrInfinity(_hour_).
+          1. Let _m_ be ! ToIntegerOrInfinity(_min_).
+          1. Let _s_ be ! ToIntegerOrInfinity(_sec_).
+          1. Let _milli_ be ! ToIntegerOrInfinity(_ms_).
+          1. Return ((𝔽(_h_) × 𝔽(msPerHour) + 𝔽(_m_) × 𝔽(msPerMinute)) + 𝔽(_s_) × 𝔽(msPerSecond)) + 𝔽(_milli_).
```

Comment:
Is it necessary to do all this arithmetic in the float space instead of doing it in the real space and then converting to float at the end?

Reply:
I don't know if it's necessary, but the spec seems to think it is. All the way back in the first edition, that step said "performing the arithmetic according to IEEE 754 rules (that is, as if using the ECMAScript operators `*` and `+`)". That persisted until PR #3091, which removed the in-step wording, but added the Note pointing out that "The arithmetic in MakeTime is floating-point arithmetic".

If one arg was a large positive and another was a large negative, maybe you could force an intermediate step to overflow if done in float-space, but not if done in real-space and then converted to float at the end?

Reply:
Since actual implementations are going to use floats, best to leave this in float space so that readers of the spec don't have to figure that out.